src/taxonomy/manage-orgs/ManageOrgsModal.jsx

```diff
@@ -0,0 +1,240 @@
+      isOpen={isOpen}
+      onClose={onClose}
+      variant="warning"
+      icon={Warning}
+      footerNode={(
+        <ActionRow>
+          <Button variant="tertiary" onClick={onClose}>
+            {intl.formatMessage(messages.cancelButton)}
+          </Button>
+          <Button variant="primary" onClick={confirm}>
+            {intl.formatMessage(messages.continueButton)}
+          </Button>
+        </ActionRow>
+      )}
+    >
+      <p>
+        {intl.formatMessage(messages.confirmUnassignText, { taxonomyName })}
+      </p>
+    </AlertModal>
+  );
+};
+
+ConfirmModal.propTypes = {
+  isOpen: PropTypes.bool.isRequired,
+  onClose: PropTypes.func.isRequired,
+  confirm: PropTypes.func.isRequired,
+  taxonomyName: PropTypes.string.isRequired,
+};
+
+const ManageOrgsModal = ({
+  taxonomyId,
+  isOpen,
+  onClose,
+}) => {
+  const intl = useIntl();
+  const { setToastMessage } = useContext(TaxonomyContext);
+
+  const [selectedOrgs, setSelectedOrgs] = useState(/** @type {null|string[]} */(null));
+  const [allOrgs, setAllOrgs] = useState(/** @type {null|boolean} */(null));
+
+  const [isConfirmModalOpen, openConfirmModal, closeConfirmModal] = useToggle(false);
+
+  const [isDialogDisabled, disableDialog, enableDialog] = useToggle(false);
+
+  const {
+    data: organizationListData,
+  } = useOrganizationListData();
+
+  const taxonomy = useTaxonomyDetailDataResponse(taxonomyId);
+
+  const manageOrgMutation = useManageOrgs();
+
+  const saveOrgs = async () => {
+    disableDialog();
+    closeConfirmModal();
+    if (selectedOrgs !== null && allOrgs !== null) {
+      try {
+        await manageOrgMutation.mutateAsync({
+          taxonomyId,
+          orgs: allOrgs ? undefined : selectedOrgs,
+          allOrgs,
+        });
+        if (setToastMessage) {
+          setToastMessage(intl.formatMessage(messages.assignOrgsSuccess));
+        }
+      } catch (/** @type {any} */ error) {
+        // ToDo: display the error to the user
+      } finally {
+        enableDialog();
+        onClose();
+      }
+    }
+  };
+
+  const confirmSave = async () => {
+    if (!selectedOrgs?.length && !allOrgs) {
+      openConfirmModal();
+    } else {
+      await saveOrgs();
+    }
+  };
+
+  useEffect(() => {
+    if (taxonomy) {
+      if (selectedOrgs === null) {
+        setSelectedOrgs([...taxonomy.orgs]);
+      }
+      if (allOrgs === null) {
+        setAllOrgs(taxonomy.allOrgs);
+      }
+    }
+  }, [taxonomy]);
+
+  useEffect(() => {
+    if (selectedOrgs) {
+      // This is a hack to force the Form.Autosuggest to clear its value after a selection is made.
+      const inputRef = /** @type {null|HTMLInputElement} */ (document.querySelector('.pgn__form-group input'));
+      if (inputRef) {
+        //  @ts-ignore
+        inputRef.value = null;
+        const event = new Event('change', { bubbles: true });
+        inputRef.dispatchEvent(event);
+      }
+    }
+  }, [selectedOrgs]);
+
```

Comment:
This seems quite hacky, but I'm not sure what alternative there is. 

That said, I think the selector here is a bit broad, I think you should scope this to `.manage_orgs`.

Reply:
Added .manage-orgs: 3cf5758fb86d9e923b617718df046d5622377e6e

There is some effort to refactor the Autosuggest component here: https://github.com/openedx/paragon/pull/2899
But for now, this is what we have :frowning:

Reply:
Good to know there some work towards a better alternative on the Paragon front!